registry.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oras
+
+import (
+	"context"
+
+	"oras.land/oras-go/v2/registry"
+)
+
+// Tag fetches the manifest identified by the srcRef, then
+// pushes the manifest with the dstRef
+func Tag(ctx context.Context, target Target, srcRef, dstRef string) error {
```

Comment:
@shizhMSFT Maybe I am missing something here, but within the scope of retagging, why can't we just use `Resolve` and `Tag` here? The `ReferenceFetcher` uses GET, which requires more resource than `Resolve` (HEAD), and we always need two requests for the retagging

Reply:
> @shizhMSFT Maybe I am missing something here, but within the scope of retagging, why can't we just use `Resolve` and `Tag` here? The `ReferenceFetcher` uses GET, which requires more resource than `Resolve` (HEAD), and we always need two requests for the retagging

The current workflow is Resolve + Tag, where Tag is Fetch + push. Therefore, current workflow may take 3 requests in total.

Reply:
@qweeah The `Tag`-manifest-descriptor functionality is not provided by the registry API. To `Tag`, the entire manifest is required in additional to the manifest descriptor. That's why we need to `GET` the manifest instead of `HEAD`.